cylc/flow/scheduler.py

```diff
@@ -1506,7 +1534,9 @@ async def main_loop(self) -> None:
 
             # Useful for debugging core scheduler issues:
             # self.pool.log_task_pool(logging.CRITICAL)
-
+            if self.incomplete_ri_map:
+                for _, platform in self.incomplete_ri_map.copy().items():
```

Comment:
Maybe consider moving this for loop down into `manage_remote_init`, could avoid having to extract the `instyall target` from the platform dict in the process.